src/ffi/socket.rs

```diff
@@ -273,7 +273,7 @@ pub struct Socket {
 impl Socket {
     /// Spawns a new [Socket] that must be [Socket::connect]ed.
     #[uniffi::constructor]
-    pub fn spawn(mut url: Url) -> Result<Arc<Self>, SpawnError> {
+    pub fn spawn(mut url: Url, cookies: Option<Vec<String>>) -> Result<Arc<Self>, SpawnError> {
```

Comment:
All the examples show this being set to `None`. Should it have a default value of `None`?

Reply:
Yes but Rust doesn't have default value's for argument parameters. I'm not sure why but I bet there's a reason. The more common pattern to have default parameters is by doing a builder pattern.